### PR TITLE
Update buildtools version to fix VS2015 build

### LINKED ISSUE #1033: Tests fail to build with VS2015
Run: build.cmd vs2015

C:\coreclr\tests\src\packages\Microsoft.DotNet.BuildTools.1.0.21-prerelease\lib\packageresolve.targets(25,5): error MSB4062: The "ResolveNuGetPackages" task could not be loaded from the assembly C:\coreclr\tests\src\packages\Microsoft.DotNet.BuildTools.1.0.21-prerelease\lib\Microsoft.DotNet.Build.Tasks.dll. Could not load file or assembly 'Microsoft.Build.Utilities.v12.0, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\coreclr\tests\src\Threading\ThreadStatics\cs_template.proj]

### PR DESCRIPTION
**Pull request by jkotas**
Fix #1033

### CODE REVIEW

**Comment by jkotas** [comment]
@ericeil @FiveTimesTheFun Could you please review?

File: tests/publishdependency.targets [hunk 1]
```diff
@@ -28,20 +28,19 @@
...
-                                Language="$(Language)"
-                                PackageRoot="$(PackagesDir)"
-                                ProjectLockFile="%(ProjectLockJsonFiles.Identity)"
-                                TargetFrameworkMonikers="@(TestTargetFramework)"
-                                TargetPlatformMonikers="$(TargetPlatformMoniker)">
-
+    <PrereleaseResolveNuGetPackageAssets AllowFallbackOnTargetSelection="true"
+                                         IncludeFrameworkReferences="false"
+                                         NuGetPackagesDirectory="$(PackagesDir)"
+                                         RuntimeIdentifier="dotnet"
```
Comment (jkotas) [inline tests/publishdependency.targets]: Is there a better value to use here instead of hardcoded dotnet?

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by ericeil** [comment]
@jkotas, did you mean to mention @ericstj?

**Comment by jkotas** [comment]
Yes, I meant @ericstj. It is fixed up already ... sorry about false alarm.

**Comment by mmitche** [comment]
LGTM

**Pull request merged**